Automated Smoketest: :x:
OS: centos 7.6.1810 (Linux 3.10.0-957.1.3.el7.x86_64)
 Host: ip-10-20-0-231.ca-central-1.compute.internal
 GCC:  gcc (GCC) 7.3.1 20180303 (Red Hat 7.3.1-5)
 Git: 2.9.5,  CMake: 3.22.1,  cUrl: 7.67.0,  node.js: v16.13.1,  npm: 8.1.2
Sha: 762368f14becc7b59a3348c3c27976db79ad014f
Containerized:False
Number of Error(s): 165
org.apache.fop.apps.FOPException: org.apache.fop.fo.ValidationException: Property ID SysAdm_OtherEnv.conf (found on fo:block) previously used; ID values must be unique within a document! (See position 838:230)
javax.xml.transform.TransformerException: org.apache.fop.fo.ValidationException: Property ID SysAdm_OtherEnv.conf (found on fo:block) previously used; ID values must be unique within a document! (See position 838:230)
Caused by: javax.xml.transform.TransformerException: org.apache.fop.fo.ValidationException: Property ID SysAdm_OtherEnv.conf (found on fo:block) previously used; ID values must be unique within a document! (See position 838:230)
Caused by: org.apache.fop.fo.ValidationException: Property ID SysAdm_OtherEnv.conf (found on fo:block) previously used; ID values must be unique within a document! (See position 838:230)
javax.xml.transform.TransformerException: org.apache.fop.fo.ValidationException: Property ID SysAdm_OtherEnv.conf (found on fo:block) previously used; ID values must be unique within a document! (See position 838:230)
Caused by: org.apache.fop.fo.ValidationException: Property ID SysAdm_OtherEnv.conf (found on fo:block) previously used; ID values must be unique within a document! (See position 838:230)
org.apache.fop.fo.ValidationException: Property ID SysAdm_OtherEnv.conf (found on fo:block) previously used; ID values must be unique within a document! (See position 838:230)
make[2]: *** [docs/EN_US/HPCCSystemAdmin/CMakeFiles/HPCCSystemAdministratorsGuide_EN_US-8.6.11-closedown0.pdf.dir/build.make:74: RelWithDebInfo/docs/EN_US/HPCCSystemAdministratorsGuide_EN_US-8.6.11-closedown0.pdf] Error 1
checking for memory.h... make[1]: *** [CMakeFiles/Makefile2:19858: docs/EN_US/HPCCSystemAdmin/CMakeFiles/HPCCSystemAdministratorsGuide_EN_US-8.6.11-closedown0.pdf.dir/all] Error 2
make: *** [Makefile:166: all] Error 2
Get ECLWatch build result
ReportTimes.
Finish at: 2022-03-14_19-46-46
time logs started
Prep time   :    19 sec (00:00:19)
Build time  :   163 sec (00:02:43)
Package time:     0 sec (00:00:00)
Install time:     0 sec (00:00:00)
Start time  :     0 sec (00:00:00)
Test time   :     0 sec (00:00:00)
Stop time   :     0 sec (00:00:00)
Summary     :   182 sec (00:03:02)
time logs end

[Build error:](https://gist.github.com/d3c4a60313e6ffb645a3abdf370c1699)
Build log [Build log](https://gist.github.com/HPCCSmoketest/d3c4a60313e6ffb645a3abdf370c1699#file-result-22-03-14-19-43-28-log)

--- a/docs/EN_US/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml
+++ b/docs/EN_US/HPCCSystemAdmin/HPCCSystemAdministratorsGuide.xml
@@ -1557,6 +1557,61 @@ lock=/var/lock/HPCCSystems</programlisting>
           </variablelist></para>
       </sect2>
 
+      <sect2 id="SysAdm_Java_env.conf">
+        <title>Environment.conf and Java Settings</title>
+
+        <para>Specify the JNI_PATH setting if HPCC cannot find the adequate
+        JNI library (libjvm) or if you want an alternate libjvm. For
+        example:</para>
+
+        <para>
+          <programlisting>JNI_PATH=/absolute/path/to/alternative/libjvm.so
+</programlisting>
+        </para>
+
+        <para>
+          <informaltable colsep="1" frame="all" rowsep="1">
+            <?dbfo keep-together="always"?>
+
+            <tgroup cols="2">
+              <colspec colwidth="49.50pt" />
+
+              <colspec />
+
+              <tbody>
+                <row>
+                  <entry>
+                    <inlinegraphic fileref="images/OSSgr3.png" />
+                  </entry>
+
+                  <entry>
+                    <para>When targeting custom classes, all dependencies must
+                    be accessible by declaring their location on the
+                    classpath, or including in the host jar file.</para>
+
+                    <para>Custom java classes should be built using compatible
+                    Java versions. To check the Java version used to build a
+                    given class use:</para>
+
+                    <para>
+                      <programlisting>javap: javap -verbose MyClass | grep "major" </programlisting>
+                    </para>
+
+                    <para>When targeting a given method, ensure the signature
+                    declared in the embedded Java section matches the actual
+                    method signature. To get the signature, use:</para>
+
+                    <para>
+                      <programlisting>javap: javap –s MyClass</programlisting>
+                    </para>
+                  </entry>
+                </row>
+              </tbody>
+            </tgroup>
+          </informaltable>
+        </para>
+      </sect2>
+
       <sect2 id="ConfiguringRemoteAccessOverTLS">
         <title>Remote Access over TLS</title>
 

